pkg/apis/agones/v1/gameserver.go

```diff
@@ -259,13 +264,29 @@ func (gs *GameServer) ApplyDefaults() {
 
 // ApplyDefaults applies default values to the GameServerSpec if they are not already populated
 func (gss *GameServerSpec) ApplyDefaults() {
+	gss.applySpecDefaults()
 	gss.applyContainerDefaults()
 	gss.applyPortDefaults()
 	gss.applyHealthDefaults()
 	gss.applySchedulingDefaults()
 	gss.applySdkServerDefaults()
 }
 
+// applySpecDefaults applies the PodTemplateSpec defaults
+func (gss *GameServerSpec) applySpecDefaults() {
+	if gss.Template.Spec.RestartPolicy == "" {
+		// TODO: Check if that is what the Agones team is expecting.
+		// Options: Always, OnFailure or Never https://pkg.go.dev/k8s.io/api/core/v1@v0.15.11?tab=doc#RestartPolicy
+		gss.Template.Spec.RestartPolicy = corev1.RestartPolicyOnFailure
+	}
+
+	if gss.Template.Spec.DNSPolicy == "" {
+		// TODO: Check if that is what the Agones team is expecting.
+		// Options: ClusterFirstWithHostNet, ClusterFirst, Default or None https://pkg.go.dev/k8s.io/api/core/v1@v0.15.11?tab=doc#DNSPolicy
+		gss.Template.Spec.DNSPolicy = corev1.DNSDefault
```

Comment:
Which DNSPolicy is expected to be used by default?

Reply:
Whatever the default is for Kubernetes +1